app/code/Magento/PageCache/etc/varnish6.vcl

```diff
@@ -113,51 +110,38 @@ sub vcl_recv {
         #unset req.http.Cookie;
     }
 
-    # Bypass authenticated GraphQL requests without a X-Magento-Cache-Id
-    if (req.url ~ "/graphql" && !req.http.X-Magento-Cache-Id && req.http.Authorization ~ "^Bearer") {
+    # Don't cache the authenticated GraphQL requests
+    if (req.url ~ "/graphql" && req.http.Authorization ~ "^Bearer") {
         return (pass);
     }
```

Comment:
This is why X-Magento-Cache-Id exists in the first place.  We need to have authorized GraphQL requests cached.

Keep in mind what the session factors currently are that create X-Magento-Cache-Id (and are able to be extended):
- Currency
- Store
- Customer Group (this won't matter anymore if we only cache the requests of guests)
- Customer tax rate
- Is logged in (this won't matter anymore if we only cache the requests of guests)

Reply:
@danmooney2 my concern is that this wil create too many cache variations which consume space in the cache and impact the hit rate.

Can you give me your opinion on my assumption about having too many cache variations?

For a typical store, how many different `X-Magento-Cache-Id` values do you think there are?

If the average number of variations would be manageable, we can choose to revert this change.

@peterjaap can you chime in as well please?

Reply:
The maximum number of variations for X-Magento-Cache-Id (without being extended) can be determined by the following equation:

Number of Currencies * Number of Stores * Number of Customer Groups * Number of Customer Tax Rates * 2 (is logged in or not).

The point is, it's obvious from current implementation that X-Magento-Cache-Id is meant to provide **both** unauthorized and authorized consumers cached responses.  

Correct me if I'm wrong, but the issue seems to be a cost benefit analysis of space consumed versus UX.  I had initially assumed this change was done to ensure an authorized user doesn't see something they weren't supposed to.  Thanks for clarifying your concern.

I'm paging GraphQL performance Product Manager @cpartica for visibility and for his take on this change.

Reply:
Curious to hear from @cpartica :)

Reply:
Yes sorry Cris is usually very busy and we don't want to hold you up.  Is this the last piece of review feedback?

Reply:
@danmooney2 I'd be happy to wait for Cris since this isn't my area of expertise since we don't work with GraphQL so much.

I'm also waiting on feedback here https://github.com/magento/magento2/pull/36796#discussion_r1126513787

Reply:
As far as I know; In concept this serves the same purpose as the `X-Magento-Vary` header. It creates separate cache buckets for the the session factors. I don't think this is a regression compared to how html pages are handled. I think the amount of buckets will be way less in practice, though, as quite a few will overlap.

Reading deeper into this, an unauthorized request ([example](https://backend.reachdigital.dev/graphql?query=query%20Layout%28%24_v1_pageSize%3AInt%24_v2_currentPage%3AInt%29%7B__typename%20categories%28pageSize%3A%24_v1_pageSize%20currentPage%3A%24_v2_currentPage%29%7B__typename%20items%7Buid%20include_in_menu%20name%20url_path%20level%20children%7Buid%20include_in_menu%20name%20url_path%20level%20children%7Buid%20include_in_menu%20name%20url_path%20level%20children%7Buid%20include_in_menu%20name%20url_path%20level%20__typename%7D__typename%7D__typename%7D__typename%7D%7D%7D&variables=%7B%22_v1_pageSize%22%3A20%2C%22_v2_currentPage%22%3A1%7D&operationName=Layout)) will return a `x-magento-cache-id` header.

So how I now understand this, there are now two systems?
1. Store + Content-Current
2. X-Magento-Cache-Id

Does this mean that in practice we should make sure we do _not_ use the first option because that will create separate cache groups?

Reply:
@IvanChepurnyi @JeroenBoersma thoughts?

Reply:
It's quite simple in my opinion, you should never cache authenticated requests based on different parameters...
So, if a Bearer token is set, the backend should always check the request.

It is authenticated for a reason, that's why the `X-Magento-Cache-Id` doesn't matter anymore.

Varnish is not for authentication, the backend is, it can check en re-evaluate the request.

You could even run into [cache poisoning](https://owasp.org/www-community/attacks/Cache_Poisoning) because of the current implementation.

Reply:
@JeroenBoersma That makes some sense, but the regular frontend requests do the same thing. It will create a different cache entry for different customer groups if one is logged in. Is your idea that this shouldn't happen at all?